tools/mesh-generators/generate_halton_mesh.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+
+import argparse
+
+import numpy as np
+import vtk
+from scipy.stats.qmc import Halton
+from vtk.util.numpy_support import numpy_to_vtk as n2v
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Generate a mesh in unit square or cube, based on Halton sequence"
+    )
+    parser.add_argument(
+        "--mesh",
+        "-m",
+        dest="output",
+        required=True,
+        help="The name of the file to create. Must be a .vtk or .vtu file.",
+    )
+    parser.add_argument(
+        "--numpoints",
+        "-n",
+        type=int,
+        dest="num_points",
+        default="100",
+        help="Number of points to generate.",
+    )
+    parser.add_argument(
+        "--dimension",
+        "-d",
+        type=int,
+        dest="dimension",
+        default="2",
+        help="Dimension of mesh (2D or 3D).",
+    )
+
+    args, _ = parser.parse_known_args()
+    return args
+
+
+def generate_points(dim, num_points):
+    points = Halton(d=dim, scramble=False).random(num_points)
```

Comment:
Does it mean the call / points we get here are non deterministic?

Reply:
The seed is generated randomly. However, it is possible to give a specific seed to the Halton function so it creates the same points for all runs.

See, https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.qmc.Halton.html

Reply:
Running the generator with the same arguments should result in the same mesh (for reproducibility). We should at least add an option for something like that.

Reply:
The seed option added 👍🏼